src/content/docs/en/reference/api-reference.mdx

```diff
@@ -1167,314 +614,321 @@ The route pattern responsible for generating the current page or route. In file-
 
 You can use this property to understand which route is rendering your component. This allows you to target or analyze similarly-generated page URLs together. For example, you can use it to conditionally render certain information, or collect metrics about which routes are slower.
 
-See also: [`Astro.routePattern`](#astroroutepattern)
-
-### `context.isPrerendered`
+### `cookies`
 
 <p>
 
-**Type**: `boolean`<br />
-<Since v="5.0.0" />
+**Type:** `AstroCookies`<br />
+<Since v="1.4.0" />
 </p>
 
-A boolean representing whether or not the current page is prerendered.
-
-You can use this property to run conditional logic in middleware, for example, to avoid accessing headers in prerendered pages.
-
-See also: [`Astro.isPrerendered`](#astroisprerendered)
+`cookies` contains utilities for reading and manipulating cookies for [routes rendered on demand](/en/guides/on-demand-rendering/).
 
-## `getStaticPaths()`
+#### Cookie utilities
 
-**Type:** `(options: GetStaticPathsOptions) => Promise<GetStaticPathsResult> | GetStaticPathsResult`
+##### `get`
 
-If a page uses dynamic params in the filename, that component will need to export a `getStaticPaths()` function.
+<p>
 
-This function is required because Astro is a static site builder. That means that your entire site is built ahead of time. If Astro doesn't know to generate a page at build time, your users won't see it when they visit your site.
+**Type:** <code>(key: string, options?: <a href="#astrocookiegetoptions">AstroCookieGetOptions</a>) => <a href="#astrocookie">AstroCookie</a> | undefined</code>
+</p>
 
-```astro
----
-export async function getStaticPaths() {
-  return [
-    { params: { /* required */ }, props: { /* optional */ } },
-    { params: { ... } },
-    { params: { ... } },
-    // ...
-  ];
-}
----
-<!-- Your HTML template here. -->
-```
+Gets the cookie as an [`AstroCookie`](#astrocookie) object, which contains the `value` and utility functions for converting the cookie to non-string types.
 
-The `getStaticPaths()` function should return an array of objects to determine which paths will be pre-rendered by Astro.
+##### `has`
 
-It can also be used in static file endpoints for [dynamic routing](/en/guides/endpoints/#params-and-dynamic-routing).
+<p>
 
-:::tip
-When using TypeScript, use the [`GetStaticPaths`](/en/guides/typescript/#infer-getstaticpaths-types) type utility to ensure type-safe access of your `params` and `props`.
-:::
+**Type:** <code>(key: string, options?: <a href="#astrocookiegetoptions">AstroCookieGetOptions</a>) => boolean</code>
+</p>
 
-:::caution
-The `getStaticPaths()` function executes in its own isolated scope once, before any page loads. Therefore you can't reference anything from its parent scope, other than file imports. The compiler will warn if you break this requirement.
-:::
+Whether this cookie exists. If the cookie has been set via `Astro.cookies.set()` this will return true, otherwise it will check cookies in the `Astro.request`.
```

Comment:
```suggestion
Whether this cookie exists. If the cookie has been set via `Astro.cookies.set()` this will return true, otherwise, it will check cookies in the `Astro.request`.
```